lightkurve/targetpixelfile.py

```diff
@@ -190,6 +195,88 @@ def from_archive(target, cadence='long', quarter=None, month=None,
     def __repr__(self):
         return('KeplerTargetPixelFile Object (ID: {})'.format(self.keplerid))
 
+
+    def get_sources(self, catalog=None, magnitude_limit=18, dist_tolerance=3):
+        """
+        Returns a crossmatched table of stars that are centered on the target
+        of the tpf file. Current catalog supported are KIC, EPIC and Gaia DR2.
+
+        Parameters
+        -----------
+        catalog: string
+            Indicate catalog assigned for mission. If Kepler, catalog will be KIC
+            if K2 catalog is EPIC.
+        magnitude_limit : int or float
+            Limit the returned magnitudes to only stars brighter than magnitude_limit.
+            Default 18.
+        dist_tolerance: float
+            Tolerance for source matching in pixels. Sources that fall within this
+            tolerance of pixels in the KeplerTargetPixelFile will be returned in the result.
+            Default is 3 pixels (12 arcsec).
+
+        Returns
+        -------
+        result : astropy.table
+            Astropy table with the following columns
+        ID : astropy.table.column (KIC & EPIC)
+            Catalog ID from catalog.
+        RAJ200: astropy.table.column (KIC & EPIC)
+            Right ascension [degrees]
+        DEJ2000: astropy.table.column (KIC & EPIC)
+            Declination [deg]
+        pmRA: astropy.table.column (KIC & EPIC)
+            Proper motion for right ascension [mas/year]
+        pmDEC: astropy.table.column (KIC & EPIC)
+        Kpmag: astropy.table.column (KIC & EPIC)
+            Magnitude in Kepler band [mag]
+        """
+
+        if catalog is None:
+            if self.mission == 'Kepler':
+                catalog = 'KIC'
+            elif self.mission == 'K2':
+                catalog = 'EPIC'
+            else:
+                log.error('Please provide a catalog.')
+
+        #Skycoord the centre of target
+        cent = SkyCoord(ra=self.ra, dec=self.dec, frame='icrs', unit=(u.deg, u.deg))
+
+        #Find the size of the TPF
+        radius = np.max(self.flux.shape[1:]) * 4 * u.arcsecond
+
+        # query around centre with radius
+        data = query_catalog(cent, radius=radius, catalog=catalog)
+
+        # Load ra & dec of all tpf pixels
+        pixels_ra, pixels_dec = self.get_coordinates(cadence='all')
+
+        # Reduce calculation for astroy seperation
+        pixel_radec = np.asarray([pixels_ra.ravel(), pixels_dec.ravel()])
+        pixel_radec = np.round(pixel_radec, decimals=5)
```

Comment:
Is rounding necessary?

Reply:
Yes, this allows us to do a `np.unique(pixel_radec)` a few lines later and find all the unique values rounded to ~0.1 arcseconds. This reduces the number of RA/Dec pairs to check from ~50,000 to ~500.

Reply:
Ah!  That means the rounding step can be dropped if we use `get_coordinates(cadence=0)` instead of `get_coordinates(cadence='all')` above.

Reply:
We want to include the motion, incase there is a nearby source that is falling on/off the detector, which would mess up our PRF photometry. We use all the RA/Dec pairs from all cadences, reduce the precision of them to 0.1 arcseconds and then check to see if any targets are "near" to any of these pixels. Reducing the precision reduces the time needed to do the check.